src/pages-and-resources/discussions/app-config-form/apps/shared/DiscussionRestriction.test.jsx

```diff
@@ -0,0 +1,195 @@
+import { generateProvidersApiResponse, legacyApiResponse } from '../../../factories/mockApiResponses';
+import OpenedXConfigForm from '../openedx/OpenedXConfigForm';
+import { selectApp } from '../../../data/slice';
+
+const courseId = 'course-v1:edX+TestX+Test_Course';
+
+describe('Discussion Restriction', () => {
+  let axiosMock;
+  let store;
+  let container;
+
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+    store = initializeStore();
+  });
+
+  afterEach(() => {
+    axiosMock.reset();
+  });
+
+  const createComponent = (onSubmit = jest.fn(), formRef = createRef(), legacy = true) => {
+    const wrapper = render(
+      <AppProvider store={store}>
+        <IntlProvider locale="en">
+          <OpenedXConfigForm
+            onSubmit={onSubmit}
+            formRef={formRef}
+            legacy={legacy}
+          />
+        </IntlProvider>
+      </AppProvider>,
+    );
+    container = wrapper.container;
+    return container;
+  };
+
+  const mockStore = async (mockResponse) => {
+    axiosMock.onGet(getDiscussionsProvidersUrl(courseId)).reply(200, generateProvidersApiResponse());
+    axiosMock.onGet(getDiscussionsSettingsUrl(courseId)).reply(200, mockResponse);
+    await executeThunk(fetchProviders(courseId), store.dispatch);
+    await executeThunk(fetchDiscussionSettings(courseId), store.dispatch);
+    store.dispatch(selectApp({ appId: 'legacy' }));
+  };
+
+  it(
+    'Successfully displayed three types (OFF, ON and SCHEDULED) for discussion restriction and selected OFF bydefault',
+    async () => {
+      const mockData = legacyApiResponse;
+      mockData.posting_restrictions = 'disabled';
+
+      await mockStore(mockData);
+      createComponent();
+
+      expect(screen.queryByTestId('disabled')).toBeInTheDocument();
+      expect(screen.queryByTestId('enabled')).toBeInTheDocument();
+      expect(screen.queryByTestId('scheduled')).toBeInTheDocument();
+      expect(container.querySelector('[data-testid="disabled"].selected-button')).toBeInTheDocument();
+    },
+  );
+
+  it.each([
+    { eventType: 'OK', description: 'ON option, when user clicks on ok button', selectedOption: 'enabled' },
+    { eventType: 'Cancel', description: 'OFF option, when user clicks on cancel button', selectedOption: 'disabled' },
+  ])('Successfully selected %s of the popup displays under ON option.', async ({ eventType, selectedOption }) => {
+    await mockStore(legacyApiResponse);
+    createComponent();
+
+    const onButton = screen.queryByTestId('enabled');
+
+    await act(async () => { fireEvent.click(onButton); });
+
+    const eventButton = await screen.findByText(eventType);
+
+    await act(async () => { fireEvent.click(eventButton); });
+
```

Comment:
const names are not descriptive and clear.

Reply:
these names are according to UI text for better recognition of component.